db/db_compaction_test.cc

```diff
@@ -6181,125 +6181,228 @@ TEST_P(DBCompactionTestWithParam, FixFileIngestionCompactionDeadlock) {
-  SyncPoint::GetInstance()->ClearAllCallBacks();
+  void RunCompactionOverlappedWithFileIngestion() {
+    if (compaction_path_to_test_ == "AutoCompaction") {
+      TEST_SYNC_POINT("PreCompaction");
+      ResumeCompactionThread();
+      // Withouth proper range conflict check,
+      // this would have been `Status::Corruption` about overlapping ranges
+      Status s = dbfull()->TEST_WaitForCompact();
+      EXPECT_OK(s);
+    } else if (compaction_path_to_test_ == "NonRefitLevelCompactRange") {
+      CompactRangeOptions cro;
+      cro.change_level = false;
+      std::string start_key = "k1";
+      Slice start(start_key);
+      std::string end_key = "k4";
+      Slice end(end_key);
+      TEST_SYNC_POINT("PreCompaction");
+      // Withouth proper range conflict check,
+      // this would have been `Status::Corruption` about overlapping ranges
+      Status s = dbfull()->CompactRange(cro, &start, &end);
+      EXPECT_OK(s);
+    } else if (compaction_path_to_test_ == "RefitLevelCompactRange") {
+      CompactRangeOptions cro;
+      cro.change_level = true;
+      cro.target_level = 5;
+      std::string start_key = "k1";
+      Slice start(start_key);
+      std::string end_key = "k4";
+      Slice end(end_key);
+      TEST_SYNC_POINT("PreCompaction");
+      Status s = dbfull()->CompactRange(cro, &start, &end);
+      // Withouth proper range conflict check,
+      // this would have been `Status::Corruption` about overlapping ranges
+      // To see this, remove the fix AND replace
+      // `DBImpl::CompactRange:PostRefitLevel` in sync point dependency with
+      // `DBImpl::ReFitLevel:PostRegisterCompaction`
+      EXPECT_TRUE(s.IsNotSupported());
+      EXPECT_TRUE(s.ToString().find("some ongoing compaction's output") !=
+                  std::string::npos);
+    } else if (compaction_path_to_test_ == "CompactFiles") {
+      ColumnFamilyMetaData cf_meta_data;
+      db_->GetColumnFamilyMetaData(&cf_meta_data);
+      ASSERT_EQ(cf_meta_data.levels[0].files.size(), 1);
+      std::vector<std::string> input_files;
+      for (const auto& file : cf_meta_data.levels[0].files) {
+        input_files.push_back(file.name);
+      }
+      TEST_SYNC_POINT("PreCompaction");
+      Status s = db_->CompactFiles(CompactionOptions(), input_files, 1);
+      // Withouth proper range conflict check,
+      // this would have been `Status::Corruption` about overlapping ranges
+      EXPECT_TRUE(s.IsAborted());
+      EXPECT_TRUE(
+          s.ToString().find(
+              "A running compaction is writing to the same output level") !=
+          std::string::npos);
+    } else {
+      assert(false);
+    }
+  }
 
-  EXPECT_EQ("0,0,0,0,0,1,1", FilesPerLevel(0));
-}
+  void DisableSyncPoints() {
+    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->ClearAllCallBacks();
+    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
+  }
 
-TEST_F(DBCompactionTest, IngestFileOfRangeOverlappedWithOngoingRefitLevel) {
```

Comment:
I realized this UT is not needed. I was planning to use this UT to demonstrate a corruption due to
```
timeline  t1 (refitlevel)                          t2 (file ingestion)
t0   |Decides to refit from Ln+1 to Ln
t1   |Release lock in LogAndApply()
t2   |                                             [a]Mistakenly decides to output to Ln cuz t1 hasn't updated LSM tree
t3   |                                             Release lock
t4   |Acquire lock 
```

However [a] couldn't happen because file ingestion actually LogAndApply() twice, first-time in https://github.com/facebook/rocksdb/blob/7.8.fb/db/db_impl/db_impl.cc#L5858 (before deciding level) and second-time in https://github.com/facebook/rocksdb/blob/7.8.fb/db/db_impl/db_impl.cc#L5352 (after deciding level).

Due to manifest writers are in order, t1's LogAndApply() will finishes before t2's first-time LogAndApply() finishes (hence t2 deciding the output level for file ingestion).Therefore by the time t2 gets to decide the output level, it already has the updated LSM after t1 so overlap-range corruption problem should not occur. 

Another way to see this is to remove my "cfd->compaction_picker()->RegisterCompaction(c.get());" for registering RefitLevel() and run the old test `DBCompactionTest, IngestFileOfRangeOverlappedWithOngoingRefitLevel`.